### PR TITLE
Add initiator_id metadata to dispatch events

### PR DESCRIPTION
**Pull request by ScottSuarez**
Adding `initiator_id` to dispatch events. Going to use this to correlate event timing (completion time, etc) for dashboards.
**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .github/workflows/test-tgc.yml [hunk 1]
```diff
@@ -25,6 +25,8 @@ on:
       sha:
         description: "The commit SHA in magic-modules repository where the status result will be posted"
         required: true
+      initiator_id:
+        description: "Metadata about the caller who initiated the dispatch event"
```
Comment (trodge) [inline .github/workflows/test-tgc.yml:29]: Maybe reword to "Identity of the workflow dispatch caller" and possibly give example values.

**Review by trodge — CHANGES_REQUESTED** [review]

File: .github/workflows/test-tgc.yml (hunk 1 above)
Comment (ScottSuarez) [reply .github/workflows/test-tgc.yml]: Changed to `caller_id`. Avoiding build id since that suggests a parent, which I want to avoid.
